File: senegal/src/main/java/com/sante/senegal/services/interfaces/HopitalService.java
package com.sante.senegal.services.interfaces;

import com.sante.senegal.dto.HopitalDto;
import com.sante.senegal.dto.HopitalRequestDto;
import com.sante.senegal.entities.Hopital;

import java.util.List;

public interface HopitalService {
    List<HopitalDto> getAllHopitaux();
    HopitalDto getHopitalById(Long id);
    List<HopitalDto> getHopitauxByStatut(Hopital.StatutHopital statut);
    List<HopitalDto> getHopitauxByVille(String ville);
    List<HopitalDto> getHopitauxByRegion(String region);
    List<HopitalDto> searchHopitauxByNom(String nom);
    List<HopitalDto> getHopitauxByTypeEtablissement(String type);
    List<HopitalDto> getHopitauxByCapaciteMinimum(Integer capaciteMin);
    HopitalDto createHopital(HopitalRequestDto dto);
    HopitalDto updateHopital(Long id, HopitalRequestDto dto);
    void deleteHopital(Long id);
    HopitalDto changeStatutHopital(Long id, Hopital.StatutHopital nouveauStatut);
}

File: medical-appointment-frontend/src/app/core/services/hopital.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { StatutHopital, Hopital, HopitalRequest } from '../models/service-hopital-rdv-disponibite/service-hopital-rdv-disponibite.module';

@Injectable({
  providedIn: 'root'
})
export class HopitalService {
  private apiUrl = 'http://localhost:8081/api/v1/hopitaux';

  constructor(private http: HttpClient) {}

  getAllHopitaux(): Observable<Hopital[]> {
    return this.http.get<Hopital[]>(this.apiUrl);
  }

  getHopitalById(id: number): Observable<Hopital> {
    return this.http.get<Hopital>(`${this.apiUrl}/${id}`);
  }

  getHopitauxByStatut(statut: StatutHopital): Observable<Hopital[]> {
    return this.http.get<Hopital[]>(`${this.apiUrl}/statut/${statut}`);
  }

  getHopitauxByVille(ville: string): Observable<Hopital[]> {
    const params = new HttpParams().set('ville', ville);
    return this.http.get<Hopital[]>(`${this.apiUrl}/ville`, { params });
  }

  getHopitauxByRegion(region: string): Observable<Hopital[]> {
    const params = new HttpParams().set('region', region);
    return this.http.get<Hopital[]>(`${this.apiUrl}/region`, { params });
  }

  searchHopitauxByNom(nom: string): Observable<Hopital[]> {
    const params = new HttpParams().set('nom', nom);
    return this.http.get<Hopital[]>(`${this.apiUrl}/search`, { params });
  }

  createHopital(hopital: HopitalRequest): Observable<Hopital> {
    return this.http.post<Hopital>(this.apiUrl, hopital);
  }

  updateHopital(id: number, hopital: HopitalRequest): Observable<Hopital> {
    return this.http.put<Hopital>(`${this.apiUrl}/${id}`, hopital);
  }

  deleteHopital(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  changeStatutHopital(id: number, statut: StatutHopital): Observable<Hopital> {
    return this.http.patch<Hopital>(`${this.apiUrl}/${id}/statut`, { statut });
  }
}